tests/test_scheduler_worker.py

```diff
@@ -59,11 +90,18 @@ def test_passing_futures_as_args_flow(scheduler_client, matrix1, matrix2):
     future2 = scheduler_client.SubmitTask(Task(taskId="1", taskDefintion="mat_add", taskData=[json.dumps(matrix1).encode(), json.dumps(matrix2).encode()]))
     future3 = scheduler_client.SubmitTask(Task(taskId="2", taskDefintion="mat_subtract", taskData=[future, future2]))
 
-    workerClient = WorkerClient(future.hostName, future.port)
-    result = json.loads(workerClient.GetResult(future3))
+    workerClient = WorkerClient(future3.hostName, future3.port)
 
     expected_result = [[13, 14], [33, 38]]
+    while True:
```

Comment:
Please extract this polling logic out to a separate function? Eliminates duplicate code.